agent/api/task.go

```diff
@@ -286,14 +291,19 @@ func (task *Task) addNetworkResourceProvisioningDependency(cfg *config.Config) {
 		if container.IsInternal() {
 			continue
 		}
-		if container.TransitionDependencySet.ContainerDependencies == nil {
-			container.TransitionDependencySet.ContainerDependencies = make([]ContainerDependency, 0)
-		}
-		container.TransitionDependencySet.ContainerDependencies = append(container.TransitionDependencySet.ContainerDependencies, ContainerDependency{
+		contDep := ContainerDependency{
 			ContainerName:   PauseContainerName,
 			SatisfiedStatus: ContainerResourcesProvisioned,
-			DependentStatus: ContainerPulled,
-		})
+		}
+		if container.TransitionDependenciesMap == nil {
+			container.TransitionDependenciesMap = make(map[ContainerStatus]TransitionDependencySet)
+		}
+		if _, ok := container.TransitionDependenciesMap[ContainerPulled]; !ok {
```

Comment:
It would be good if you can wrap this as a method of container struct. So you can use it in both the `empty volume` and the `pause container`.